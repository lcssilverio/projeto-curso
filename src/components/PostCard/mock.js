export const mockPostCard = {
  title: "Title",
  body: "Body",
  id: 1,
  cover: "img/img.png",
};
