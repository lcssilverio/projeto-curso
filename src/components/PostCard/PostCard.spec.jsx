import { render, screen } from "@testing-library/react"
import { PostCard } from '.'
import { mockPostCard } from "./mock"

describe('Tests to PostCard component', () => {
  test('should render the PostCard correctly', () => {
    render(<PostCard {...mockPostCard} />)
    expect(screen.getByRole('img', {name: 'Title'}))
      .toHaveAttribute('src', 'img/img.png')
    expect(screen.getByRole('heading', {name: 'Title 1'})).toBeInTheDocument()
    expect(screen.getByText('Body')).toBeInTheDocument()
  })

  test('should match snapshot', () => {
    const {container} = render(<PostCard {...mockPostCard} />)
    expect(container.firstChild).toMatchSnapshot()
  })

})
