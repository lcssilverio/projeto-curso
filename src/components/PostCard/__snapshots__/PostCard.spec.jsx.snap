// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests to PostCard component should match snapshot 1`] = `
<div>
  <div
    class="post"
  >
    <img
      alt="Title"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Title
         
        1
      </h2>
      <p>
        Body
      </p>
    </div>
  </div>
</div>
`;
